enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,587 @@
+
+
+## Proposal
+
+The proposal commences with the acknowledgement that a user requires an ETCD backup.
+
+The primary requirement is that the ETCD `managementType` must be configured as `managed`. Once this configuration is set, further progress can proceed.
+
+This backup operation will be specified within the `hostedcluster.spec.etcd` API group under a new subgroup named `backup`. This subgroup will encompass two fields, with the primary one designated for the backup process itself, directly imported from the original ETCD Backup API. Comprehensive details regarding this API can be found in the official ETCD proposal and its corresponding implementation documentation.
+
+- [ETCD Operator Backup API Design Doc](https://docs.google.com/document/d/1xBnK3D7MCblFgxQ8RUZFyWNIU_bnSFHciLhhHAeP0Xk/edit)
+- [ETCD Backup API implementation](https://github.com/openshift/api/pull/1482)
+- [Periodic Backup Controller](https://github.com/openshift/cluster-etcd-operator/pull/1074)
+
+The other field it's called `storeEndpoint` which basically will take care of the export of the binary to an external endpoint. This endpoint could be S3, Azure, etc.... Inside of this `storeEndpoint` we have some fields:
+
+- **BackupPrefix:** Where the user could set some kind of prefix to identify that backup in a more readable way. This Prefix will be parsed, validated and used in the final name which will contain the **timestamp**, the **HostedCluster name** and the provided **BackupPrefix**.
+- **Credentials:** This is a `SecretReference` located in the HostedCluster namespace using the same name as the backup plus the "credentials" suffix
+- **ServiceConfig:** This ServiceConfig field is another struct which it will contain the details about the remote destination you will upload that backup.
+
+The `ServiceConfig` struct will be the same on each provider. The plan is to provide a common interface and modify the behaviour depending on the destination provider.
+
+- **ServiceURL:** Points to the URL necessary to upload the binary.
+- **Region:** This will set the Region for the destination bucket.
+- **Tags:** Necessary tags to label the backup in he destination resources.
+
+### Workflow Description
+
+#### Backup
+
+When a user with a HostedCluster intends to perform a backup, the initial step involves modifying the HostedCluster by introducing a new field within the etcd API group. This new field will serve to incorporate the required parameters and configurations essential for executing backup operations seamlessly within the HostedCluster environment.
+
+```yaml
+...
+spec:
+  controllerAvailabilityPolicy: SingleReplica
+  dns:
+    baseDomain: hypershiftbm.lab
+  etcd:
+    managed:
+      storage:
+        persistentVolume:
+          size: 8Gi
+        restoreSnapshotURL: null
+        type: PersistentVolume
+    managementType: Managed
+    backup:
```

Comment:
why not specify the backup inside the `managed` struct?

Reply:
Makes sense.